clients/fides-js/src/lib/preferences.ts

```diff
@@ -68,36 +67,65 @@ export const updateConsentPreferences = async ({
         })
       : undefined;
 
-  // 1. Save preferences to Fides API
-  if (!fidesDisableSaveApi) {
-    debugLog(debug, "Saving preferences to Fides API");
-    const privacyPreferenceCreate: PrivacyPreferencesRequest = {
-      browser_identity: cookie.identity,
-      preferences: fidesUserPreferences,
-      privacy_experience_id: experienceId,
-      user_geography: userLocationString,
-      method: consentMethod,
-      ...(tcf ?? []),
-    };
-    await patchUserPreferenceToFidesServer(
-      privacyPreferenceCreate,
-      fidesApiUrl,
-      debug
-    );
-  }
-
-  // 2. Update the cookie object based on new preferences
+  // 1. Update the cookie object based on new preferences
   const updatedCookie = await updateCookie(cookie);
   Object.assign(cookie, updatedCookie);
 
-  // 3. Update the window.Fides object
-  debugLog(debug, "Updating window.Fides");
+  // 2. Update the window.Fides object
+  debugLog(fidesConfig.options.debug, "Updating window.Fides");
   window.Fides.consent = cookie.consent;
   window.Fides.fides_string = cookie.fides_string;
   window.Fides.tcf_consent = cookie.tcf_consent;
 
+  // 3. Save preferences to Fides API
+  if (!fidesConfig.options.fidesDisableSaveApi) {
+    if (fidesConfig.options.api?.savePreferencesFn) {
+      try {
```

Comment:
nit: I find this chunk of logic rather "long" and unwieldy when compared to the rest of the chunks in this `updateConsentPreferences` function. I'd recommend a few tidy-ups to look like this:
```
// 3. Save preferences to API (if not disabled)
if (!fidesConfig.options.fidesDisableSaveApi) {
  try {
    await savePreferencesApi(...);
  } catch (e) {
    debugLog(fidesConfig.options.debug, "Error saving updated preferences to API, continuing. Error: ", e);
  }
}
```

Then encapsulate the main if/else into that `savePreferencesApi` helper. This does a few things:
1. Makes this overall function much more readable, because it's effectively calling 6 helpers
2. Uses the same try/catch for both branches, I don't see much value in slightly different logs for each
3. Forces you to think about the arguments to the two branches and ideally just tighten up the code